src/providers/emoji/Emojis.cpp

```diff
@@ -296,31 +319,61 @@ std::vector<boost::variant<EmotePtr, QString>> Emojis::parse(
 
         for (const std::shared_ptr<EmojiData> &emoji : possibleEmojis)
         {
+            int emojiNonQualifiedExtraCharacters =
+                emoji->nonQualified.length() - 1;
             int emojiExtraCharacters = emoji->value.length() - 1;
-            if (emojiExtraCharacters > remainingCharacters)
+            if (remainingCharacters >= emojiExtraCharacters)
             {
-                // It cannot be this emoji, there's not enough space for it
-                continue;
-            }
+                // look in emoji->value
+                bool match = true;
 
-            bool match = true;
+                for (int j = 1; j < emoji->value.length(); ++j)
+                {
+                    if (text.at(i + j) != emoji->value.at(j))
+                    {
+                        match = false;
 
-            for (int j = 1; j < emoji->value.length(); ++j)
-            {
-                if (text.at(i + j) != emoji->value.at(j))
+                        break;
+                    }
+                }
+
+                if (match)
                 {
-                    match = false;
+                    matchedEmoji = emoji;
+                    matchedEmojiLength = emoji->value.length();
 
                     break;
                 }
             }
-
-            if (match)
+            else if (remainingCharacters >= emojiNonQualifiedExtraCharacters)
```

Comment:
This `else if` should be an `if` otherwise nonqualified emojis will never be matched in the middle of strings.

Reply:
And it should include `!emoji->nonQualified.isNull()`.

Reply:
Fixed in [`bc8e731` (#4840)](https://github.com/Chatterino/chatterino2/pull/4840/commits/bc8e7317a50217f1053dcd56d1008f6290909f27)